aten/src/ATen/native/mps/operations/Indexing.h

```diff
@@ -0,0 +1,56 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/TensorFactory.h>
+#include <c10/core/ScalarType.h>
+#include <torch/library.h>
+#include <unordered_map>
+
+#ifdef __OBJC__
+#include <MetalPerformanceShaders/MetalPerformanceShaders.h>
+#endif
+
+using namespace at::mps;
+
+namespace at {
+namespace native {
+namespace mps {
+
+std::string getMetalScalarType(ScalarType scalar_type) {
+  std::string res = "";
+  switch (scalar_type) {
+    case ScalarType::Float:
+      res = "float"; break;
+    case ScalarType::Half:
+      res = "half";  break;
+    case ScalarType::Long:
+      res = "long";  break;
+    case ScalarType::Int:
+      res = "int";   break;
+    case ScalarType::Short:
+      res = "short"; break;
+    case ScalarType::Char:
+      res = "char"; break;
+    case ScalarType::Byte:
+      res = "uchar"; break;
+    case ScalarType::Bool:
+      res = "bool";  break;
+    default:
+      break;
```

Comment:
```suggestion
    default:
      TORCH_CHECK(false, "Unsupported scalar type", scalar_type);
      break;
```

Reply:
The checks are made at the beginning of the index.Tensor_out:
```
    TORCH_CHECK(inputTensor.scalar_type() == ScalarType::Float ||
                inputTensor.scalar_type() == ScalarType::Half  ||
                inputTensor.scalar_type() == ScalarType::Long  ||
                inputTensor.scalar_type() == ScalarType::Int   ||
                inputTensor.scalar_type() == ScalarType::Short ||
                inputTensor.scalar_type() == ScalarType::Char  ||
                inputTensor.scalar_type() == ScalarType::Byte  ||
                inputTensor.scalar_type() == ScalarType::Bool, getMPSTypeString(inputTensor.scalar_type()) + std::string(" not supported for index.Tensor_out"));
```

Reply:
Thanks as this check is essentially free (not supposed to be hit ever, but better than returning an empty string (imo)